{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "eedf59cd",
   "metadata": {},
   "source": [
    "# 01 Preprocessing\n",
    "\n",
    "This notebook covers the preprocessing of the 16S rRNA amplicon sequencing data. It includes data import, quality control, denoising, and taxonomic classification.\n",
    "\n",
    "<img src=./figures/workflow_preprocessing.jpg alt=\"Description\" width=\"750\" height=\"\">"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c973fdaf",
   "metadata": {},
   "source": [
    "## Setup\n",
    "\n",
    "Activate the environment `microbEvolve` before running this Jupiter notebook. \n",
    "\n",
    "This step loads all requited packages. The paths to the scripts and the data are stored in the variables `scripts_dir` and `data_dir`. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "cd58fc87",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-12-05T14:40:44.840194Z",
     "iopub.status.busy": "2025-12-05T14:40:44.839532Z",
     "iopub.status.idle": "2025-12-05T14:40:44.844917Z",
     "shell.execute_reply": "2025-12-05T14:40:44.844376Z",
     "shell.execute_reply.started": "2025-12-05T14:40:44.840167Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "from qiime2 import Visualization\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "e322f0c5",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-12-05T14:40:45.030152Z",
     "iopub.status.busy": "2025-12-05T14:40:45.029780Z",
     "iopub.status.idle": "2025-12-05T14:40:45.033499Z",
     "shell.execute_reply": "2025-12-05T14:40:45.032982Z",
     "shell.execute_reply.started": "2025-12-05T14:40:45.030126Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "scripts_dir = \"src\"\n",
    "data_dir = \"../data\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9bd7e388",
   "metadata": {},
   "source": [
    "## Importing sequencing data and metadata\n",
    "\n",
    "This step imports the demultiplexed data, which is already available as a QIIME 2 artifact on Polybox.\n",
    "\n",
    "Next, the metadata is imported. It is provided as an Excel file with three sheets: DataDictionary, metadata_per_sample, and metadata_per_age. Each sheet is imported as its own .tsv file (`metadata_dictionary`, `metadata_per_sample`, and `metadata_per_age`). Lastly, the first column in metadata_per_sample `Unnamed: 0` is renamed to `sampleid`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ca3d8694",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2025-12-06 14:40:57] Starting importing script\n",
      "[2025-12-06 14:40:57] Importing 16S rRNA sequencing data...\n",
      "--2025-12-06 14:40:59--  https://polybox.ethz.ch/index.php/s/zi5ZBrBwcn7SYof/download/demux-paired-end.qza\n",
      "Auflösen des Hostnamens polybox.ethz.ch (polybox.ethz.ch)… 129.132.71.243\n",
      "Verbindungsaufbau zu polybox.ethz.ch (polybox.ethz.ch)|129.132.71.243|:443 … verbunden.\n",
      "HTTP-Anforderung gesendet, auf Antwort wird gewartet … 200 OK\n",
      "Länge: 572079812 (546M) [application/octet-stream]\n",
      "Wird in »../data/raw/demux_paired_end.qza« gespeichert.\n",
      "\n",
      "../data/raw/demux_p 100%[===================>] 545,58M  7,09MB/s    in 2m 48s  \n",
      "\n",
      "2025-12-06 14:43:47 (3,25 MB/s) - »../data/raw/demux_paired_end.qza« gespeichert [572079812/572079812]\n",
      "\n",
      "[2025-12-06 14:43:48] 16S rRNA sequencing import completed and stored in ../data/raw/dada2_rep_set.qza\n",
      "[2025-12-06 14:43:48] Importing metadata...\n",
      "--2025-12-06 14:43:48--  https://polybox.ethz.ch/index.php/s/YQQggAqcQCApJmQ/download\n",
      "Auflösen des Hostnamens polybox.ethz.ch (polybox.ethz.ch)… 129.132.71.243\n",
      "Verbindungsaufbau zu polybox.ethz.ch (polybox.ethz.ch)|129.132.71.243|:443 … verbunden.\n",
      "HTTP-Anforderung gesendet, auf Antwort wird gewartet … 200 OK\n",
      "Länge: 25082 (24K) [application/vnd.openxmlformats-officedocument.spreadsheetml.sheet]\n",
      "Wird in »../data/raw/metadata.xlsx« gespeichert.\n",
      "\n",
      "../data/raw/metadat 100%[===================>]  24,49K  --.-KB/s    in 0,03s   \n",
      "\n",
      "2025-12-06 14:43:48 (825 KB/s) - »../data/raw/metadata.xlsx« gespeichert [25082/25082]\n",
      "\n",
      "[2025-12-06 14:43:49] Metadata import completed and stored in ../data/raw/metadata.xlsx\n",
      "[2025-12-06 14:43:49] Saving metadata as single .tsv files...\n",
      "[2025-12-06 14:43:52] metadata_dictionary metadata_per_age and metadata_per_sample stored successfuly as single .tsv files in ../data/raw/\n",
      "[2025-12-06 14:43:52] Import script completed successfully\n"
     ]
    }
   ],
   "source": [
    "! sh {scripts_dir}/importing.sh"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4619e07f",
   "metadata": {},
   "source": [
    "## Quality Control\n",
    "\n",
    "For quality control, we converted the `demux_paired_end.qza` file into a `.qzv` artifact."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "08dd9653",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/yararoth/miniconda3/envs/microbEvolve/lib/python3.10/site-packages/emperor/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[31m\u001b[1mPlugin error from demux:\n",
      "\n",
      "  /var/folders/56/1kg2s72566n3fjbq7sy5d0mm0000gn/T/qiime2/yararoth/data/3e48be7c-af27-4f87-b7d2-2726aca3b65d/data is not a(n) SingleLanePerSamplePairedEndFastqDirFmt:\n",
      "\n",
      "  A pair of paired-end files were found not to have the same number of records. /var/folders/56/1kg2s72566n3fjbq7sy5d0mm0000gn/T/qiime2/yararoth/data/3e48be7c-af27-4f87-b7d2-2726aca3b65d/data/sample_5_S771_L001_R1_001.fastq.gz has 112892 records. /var/folders/56/1kg2s72566n3fjbq7sy5d0mm0000gn/T/qiime2/yararoth/data/3e48be7c-af27-4f87-b7d2-2726aca3b65d/data/sample_51_S266_L001_R1_001.fastq.gz has 14284 records.\n",
      "\n",
      "Debug info has been saved to /var/folders/56/1kg2s72566n3fjbq7sy5d0mm0000gn/T/qiime2-q2cli-err-svzmrumr.log\u001b[0m\n",
      "\u001b[?25h\u001b[0m"
     ]
    }
   ],
   "source": [
    "! sh {scripts_dir}/quality_control.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "ec191742",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-12-05T14:40:17.829947Z",
     "iopub.status.busy": "2025-12-05T14:40:17.829377Z",
     "iopub.status.idle": "2025-12-05T14:40:17.839658Z",
     "shell.execute_reply": "2025-12-05T14:40:17.839130Z",
     "shell.execute_reply.started": "2025-12-05T14:40:17.829923Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/85250467-d74f-413e-a4b7-890e8b206540')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 85250467-d74f-413e-a4b7-890e8b206540>"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/processed/demux_paired_end.qzv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8ec743ad",
   "metadata": {},
   "source": [
    "The minimum and maximum sequencing length during subsampling was 301 bp for both forward and reverse reads.\n",
    "\n",
    "Overall, sequence quality was very high:\n",
    "- Forward reads: median quality score of 34 across all base positions.\n",
    "- Reverse reads: median quality score of 34 up to position 293, dropping to 20 from position 294 onward.\n",
    "\n",
    "Variability was generally higher in the reverse reads and increased substantially from position 221 onward."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d71a5097",
   "metadata": {},
   "source": [
    "## Cutadapt and Denoising"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b5e0b050",
   "metadata": {},
   "source": [
    "### Cutadapt\n",
    "\n",
    "Interestingly, the V4 region of the 16S rRNA gene is much shorter than 301 bp, which is why we thought that we might still have primer sequences in our reads.\n",
    "\n",
    "#### 1. Initial Trimming Attempt (Original and Modified Primers)\n",
    "\n",
    "The first attempt used the known V4 specific forward and reverse primer sequences ([source](https://earthmicrobiome.ucsd.edu/protocols-and-standards/16s/)). We used the `--p-discard-untrimmed True` flag in order to see how many reads would be trimmed.\n",
    "- This step resulted in zero sequences being trimmed or retained.\n",
    "- In hindsight, it is irrelevant if the original or modified primers are used, as they are not required to match perfectly and the one base difference does not influence the result.\n",
    "- This indicated that the forward and reverse primers were already removed from the sequences by the sequencing facility prior to data delivery. This was also confirmed by our TA. Because we anchored the primers, the `--p-discard-untrimmed True` setting caused all reads to be discarded even if the reverse complement might be present.\n",
    "\n",
    "The other possibility that could explain the length of our reads would be read-through.\n",
    "\n",
    "#### 2. Identifying and Trimming Read-Through (Successful Strategy)\n",
    "\n",
    "To be able to identify read-through, even though the forward and reverse primers had already been removed from the sequences, we decided to only look for the reverse complement of the primers.\n",
    "- Approximately 4.5 million sequences were successfully truncated and retained using this approach, which confirms the presence of read-through.\n",
    "- The successful trim yielded reads with an approximate length of 250 bases, which likely represents the true length of the amplicon.\n",
    "- **Many reverse reads were longer** than 250 bases, suggesting that the low base quality towards the end prevented cutadapt from recognizing the reverse-complement forward primer due to many mismatches."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "4f1465f5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/yararoth/miniconda3/envs/microbEvolve/lib/python3.10/site-packages/emperor/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[31m\u001b[1mPlugin error from cutadapt:\n",
      "\n",
      "  /var/folders/56/1kg2s72566n3fjbq7sy5d0mm0000gn/T/qiime2/yararoth/data/3e48be7c-af27-4f87-b7d2-2726aca3b65d/data is not a(n) SingleLanePerSamplePairedEndFastqDirFmt:\n",
      "\n",
      "  A pair of paired-end files were found not to have the same number of records. /var/folders/56/1kg2s72566n3fjbq7sy5d0mm0000gn/T/qiime2/yararoth/data/3e48be7c-af27-4f87-b7d2-2726aca3b65d/data/sample_5_S771_L001_R1_001.fastq.gz has 112892 records. /var/folders/56/1kg2s72566n3fjbq7sy5d0mm0000gn/T/qiime2/yararoth/data/3e48be7c-af27-4f87-b7d2-2726aca3b65d/data/sample_51_S266_L001_R1_001.fastq.gz has 14284 records.\n",
      "\n",
      "Debug info has been saved to /var/folders/56/1kg2s72566n3fjbq7sy5d0mm0000gn/T/qiime2-q2cli-err-7eh3c_zw.log\u001b[0m\n",
      "\u001b[?25h\u001b[0m"
     ]
    }
   ],
   "source": [
    "! sh {scripts_dir}/cutadapt.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7fadee29",
   "metadata": {},
   "outputs": [],
   "source": [
    "Visualization.load(f\"{data_dir}/processed/demux_paired_end_trimmed-modified-primers.qza\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "669f6828",
   "metadata": {},
   "outputs": [],
   "source": [
    "Visualization.load(f\"{data_dir}/processed/demux_paired_end_trimmed-original-primers.qzv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "af1112c1",
   "metadata": {},
   "source": [
    "### Denoising with DADA2\n",
    "\n",
    "As trimming with cutadapt failed for many reverse reads, we decided to truncate aggressively during denoising to remove any read-through sequences. The truncation lengths were set to 220 bp for forward reads and 200 bp for reverse reads.\n",
    "\n",
    "A truncation length of 220 basses for the forward reads and 200 bass pairs for the reverse reads resulted in good denoising performance. Around 90% of the reads passed the filtering step and nearly all of those reads were able to be merged."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4a01dff4",
   "metadata": {},
   "outputs": [],
   "source": [
    "! sh {scripts_dir}/denoising.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "7a2412de",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/var/folders/56/1kg2s72566n3fjbq7sy5d0mm0000gn/T/qiime2/yararoth/data/9c3d2309-0af3-4b0a-9ae7-bb739f03511f')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 9c3d2309-0af3-4b0a-9ae7-bb739f03511f>"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/processed/dada2_stats.qzv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5b3df9ce",
   "metadata": {},
   "source": [
    "## Taxonomy\n",
    "\n",
    "To identify which organisms are present in our samples, we used a pretrained, weighted classifier optimized for stool samples. It targets the 16S rRNA V4 region (515F/806R) and is based on the SILVA 138.2 database (99% NR). This classifier incorporates weights derived from a large database of human stool samples. This is designed to improve classification accuracy for samples derived from the human gut by prioritizing taxa commonly found in that environment."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b96ca58a",
   "metadata": {},
   "outputs": [],
   "source": [
    "! sh {scripts_dir}/taxonomy.sh"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "microbEvolve",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
