{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "eedf59cd",
   "metadata": {},
   "source": [
    "# 01 Preprocessing"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c973fdaf",
   "metadata": {},
   "source": [
    "## Setup\n",
    "\n",
    "We stored all our shell scripts in the folder \"src\". Executing this JupiterNotebook will execute every shell script in the right order. The submission files are also awailable in the folder \"src\", in case on wants to run the shell scripts on Euler."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "cd58fc87",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-12-05T14:40:44.840194Z",
     "iopub.status.busy": "2025-12-05T14:40:44.839532Z",
     "iopub.status.idle": "2025-12-05T14:40:44.844917Z",
     "shell.execute_reply": "2025-12-05T14:40:44.844376Z",
     "shell.execute_reply.started": "2025-12-05T14:40:44.840167Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "from qiime2 import Visualization\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "e322f0c5",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-12-05T14:40:45.030152Z",
     "iopub.status.busy": "2025-12-05T14:40:45.029780Z",
     "iopub.status.idle": "2025-12-05T14:40:45.033499Z",
     "shell.execute_reply": "2025-12-05T14:40:45.032982Z",
     "shell.execute_reply.started": "2025-12-05T14:40:45.030126Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "scripts_dir = \"src\"\n",
    "data_dir = \"../data\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9bd7e388",
   "metadata": {},
   "source": [
    "## Importing sequencing data and metadata\n",
    "\n",
    "Documentation missing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ca3d8694",
   "metadata": {},
   "outputs": [],
   "source": [
    "! sh {scripts_dir}/importing.sh"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4619e07f",
   "metadata": {},
   "source": [
    "## Quality Control\n",
    "\n",
    "Documentation missing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "08dd9653",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/yararoth/miniconda3/envs/microbEvolve/lib/python3.10/site-packages/emperor/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[31m\u001b[1mPlugin error from demux:\n",
      "\n",
      "  /var/folders/56/1kg2s72566n3fjbq7sy5d0mm0000gn/T/qiime2/yararoth/data/3e48be7c-af27-4f87-b7d2-2726aca3b65d/data is not a(n) SingleLanePerSamplePairedEndFastqDirFmt:\n",
      "\n",
      "  A pair of paired-end files were found not to have the same number of records. /var/folders/56/1kg2s72566n3fjbq7sy5d0mm0000gn/T/qiime2/yararoth/data/3e48be7c-af27-4f87-b7d2-2726aca3b65d/data/sample_5_S771_L001_R1_001.fastq.gz has 112892 records. /var/folders/56/1kg2s72566n3fjbq7sy5d0mm0000gn/T/qiime2/yararoth/data/3e48be7c-af27-4f87-b7d2-2726aca3b65d/data/sample_51_S266_L001_R1_001.fastq.gz has 14284 records.\n",
      "\n",
      "Debug info has been saved to /var/folders/56/1kg2s72566n3fjbq7sy5d0mm0000gn/T/qiime2-q2cli-err-tmzxlxbi.log\u001b[0m\n",
      "\u001b[?25h\u001b[0m"
     ]
    }
   ],
   "source": [
    "! sh {scripts_dir}/quality_control.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "ec191742",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-12-05T14:40:17.829947Z",
     "iopub.status.busy": "2025-12-05T14:40:17.829377Z",
     "iopub.status.idle": "2025-12-05T14:40:17.839658Z",
     "shell.execute_reply": "2025-12-05T14:40:17.839130Z",
     "shell.execute_reply.started": "2025-12-05T14:40:17.829923Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/85250467-d74f-413e-a4b7-890e8b206540')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 85250467-d74f-413e-a4b7-890e8b206540>"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/processed/demux_paired_end.qzv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d71a5097",
   "metadata": {},
   "source": [
    "## Denoising\n",
    "Flu fragen, ob er das noch ausf√ºllen kann"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c38b80d7",
   "metadata": {},
   "source": [
    "Even though we could have used the trimmed reads from the previous step, we decided to simply truncate them aggressively in the denoising step directly.\n",
    "A truncation length of 220 basses for the forward reads and 200 bass pairs for the reverse reads resulted in good denoising performance. Around 90% of the reads passed the filtering step and nearly all of those reads were able to be merged."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4f1465f5",
   "metadata": {},
   "outputs": [],
   "source": [
    "! sh {scripts_dir}/denoising.sh"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5b3df9ce",
   "metadata": {},
   "source": [
    "## Taxonomy\n",
    "\n",
    "To assess the impact of prior knowledge on assignment accuracy, we compare two pre-trained classifiers targeting the 16S rRNA V4 region (515F/806R) from the SILVA 138.2 database (99% NR):\n",
    "- Weighted (Stool-Optimized) Classifier: This classifier incorporates weights derived from a large database of human stool samples. This is designed to improve classification accuracy for samples derived from the human gut by prioritizing taxa commonly found in that environment.\n",
    "- Unweighted Classifier: This is the standard Naive Bayes classifier, which treats all reference sequences in the database equally.\n",
    "\n",
    "We can now try to assess differences in classification between the two approaches.\n",
    "\n",
    "**We did not yet assess differences**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b96ca58a",
   "metadata": {},
   "outputs": [],
   "source": [
    "! sh {scripts_dir}/taxonomy.sh"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "QIIME 2",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
