\documentclass[a4paper, twocolumn, 10pt]{article}

\usepackage{anysize} % Package to change margin size
\marginsize{2cm}{2cm}{1cm}{2cm}
\usepackage{fancyhdr} % Package to make headers
\renewcommand{\headrulewidth}{0pt}
\usepackage[dvipsnames]{xcolor} % Colors for the references links
\usepackage{hyperref} % Package to link references
\hypersetup{
    colorlinks=true,
    linkcolor=black,
    citecolor=CadetBlue,
    filecolor=CadetBlue,      
    urlcolor=CadetBlue,
}

% \usepackage{float}
\usepackage{graphicx, subcaption, stfloats}
\usepackage{booktabs, multirow}
\usepackage{amsmath}

\usepackage[style=numeric, sorting=nyt, url=false, isbn=false, backend=biber]{biblatex}
\addbibresource{./microbEvolve2.bib}

\DeclareBibliographyDriver{misc}{%
  \printnames{author}
  \printfield{title}
  \printfield{url}%
  \printfield{urldate}
  \finentry
}

\renewenvironment{abstract} % Sets abstract
 {\par\noindent\textbf{\abstractname}\ \ignorespaces \\}
 {\par\noindent\medskip}

 
\begin{document}
% Makes header
\pagestyle{fancy}
% \thispagestyle{empty}
\fancyhead[R]{\includegraphics[height=1cm]{misc/eth_logo_kurz_pos.png}}
\fancyhead[L]{}
% Makes footnotes with an asterisk
\renewcommand*{\thefootnote}{\fnsymbol{footnote}}
    
\twocolumn[
  \begin{@twocolumnfalse}

% TITLE ----
\begin{center}
\Large{\textbf{MicrobEvolve2 -- Final Report}} \\
\vspace{0.4cm}
\normalsize
\begin{tabular}{rl}
  Yara Roth & \texttt{rothya@ethz.ch} \\
  Flurin Schindele & \texttt{fschindele@ethz.ch} \\
  Emelie Guggisberg & \texttt{eguggisberg@ethz.ch}
\end{tabular} \\
\vspace{0.1cm}
% \textit{ETH Zurich}
%\small{Other Location}
\medskip
\normalsize
\end{center}
{\color{gray}\hrule}
\vspace{0.4cm}

% ABSTRACT ----- (if needed)
\begin{abstract}

{\color{gray}\hrule}
\medskip
\end{abstract}

\end{@twocolumnfalse}
]

%\tableofcontents

% TEXT ----

% HINTS ----
% Please use the figure* (or table*) environment instead of figure to include
% page-wide graphics. Figure will use column-wide graphics.

\section*{Introduction}
During the first years of life, the gut microbiome undergoes a rapid and extensive development before stabilizing into an adult-like state \cite{milani_first_2017,fahur_bottino_early_2025,yassour_natural_2016}. This rapid development is influenced by multiple factors including the maternal microbiota, mode of delivery, feeding method and antibiotic exposure \cite{zimmermann_factors_2018,fahur_bottino_early_2025,bokulich_antibiotics_2016}. \\
The gut microbial communities play a key role in immune, metabolic, and endocrine pathways, and therefore directly influence host development \cite{robertson_human_2019,bokulich_antibiotics_2016}. This also explains why altered colonization events in early life are associated with the development of inflammatory disorders, metabolic diseases, and neurocognitive outcomes \cite{fahur_bottino_early_2025}.\\
Growing evidence suggests a bidirectional relationship between sleep and gut microbiome composition, with many sleep disorders linked to alterations in the microbiome \cite{sen_microbiota_2021}.
Identifying key microbial differences linked to neurocognitive functions, such as sleep  or rhythmicity could uncover potential therapeutic targets, for example through targeted nutritional interventions to improve sleep.\\

The goals of this project are to analyze the development of microbial diversity in infants during their first six months of life. We also aim to assess compositional changes and explore their functional implications. In a second part, we investigate how microbiome changes relate to behavioural outcomes measured and attempt to predict these outcomes from microbial profiles.\\
We worked with the data from an observational longitudinal cohort study that followed healthy infants during their first six months of life \cite{kerff_gut_2025}. Stool samples were collected at 2, 4 and 6 months, along with behavioural measures. The V4 region of the 16S rRNA gene was sequenced using Illumina NextSeq2000.

\section*{Methods}
\subsection*{Preprocessing}
The preprocessing of the 16S rRNA amplicon sequencing data includes data importing,  control, cutadapt and denoising, taxonomic classification and feature table preparation (see Figure \ref{fig:workflow_preprocessing}). \\
First, the demultiplexed sequences which were provided as QIIME 2 artifacts were imported along with the corresponding metadata.  control was performed next. Forward reads displayed a median  score of 34 for all base positions. The reverse reads also displayed a median of 34, which dropped to 20 for the last base positions. The variability was higher overall for the reverse reads and increases substantially towards the end of the reads. %(see Figure \ref{fig:_control}).
Lastly, both forward and reverse sequences displayed a read length of 301 bp. \\
The V4 region of the 16S rRNA gene is shorter than 301 bp, which indicated the presence of primers or read-through in the sequences \cite{ezbiocloud_16s_nodate}. To verify the presence of primers, we performed an initial trimming attempt with Cutadapt using V4-specific forward and reverse primer sequences \cite{martin_cutadapt_2011,earthmicrobiome_16s_nodate}. This resulted in zero sequences being trimmed, meaning that the primers were already removed by the sequencing facility. Another possible explanation for the read length was read-through, which was confirmed by running Cutadapt with the reverse complements of the primers. Approximately 4.5 million sequences were successfully truncated, resulting in reads of $\approx$250 bases, likely representing the true amplicon length. Truncation failed for many reversed reads, likely due to the low base  at the ends, preventing Cutadapt from recognizing the reverse complement primer. Therefore, truncation was perfomed directly during denoising with DADA2 \cite{callahan_dada2_2016}. Forward and reverse reads were truncated to 220 bp and 200 bp, which is sufficient to remove the read-through while maintaining an overlap of $\approx$130 base pairs. This resulted in good denoising performance, with 90\% of reads passing filtering and nearly all reads successfully merged. \\
Next, taxanomic classification was performed using a weighted classifier optimized for stool samples \cite{bokulich_optimizing_2018,quast_silva_2013}. The classifier targets the 16S rRNA V4 region (515F/806R) and is based on the SILVA 138.2 database (99\% NR) . \\ 
The last step of preprocessing involved preparation of the feature table and metadata. The metadata consist of two files: one with per sample information and one with per-age information, including behavioural outcome measures at each timepoint. Both files were merged into a complete metadata file to simplify further analysis.\\
The number of samples collected per infant at a given timepoint varies depending on stool frequency, resulting in uneven sampling across infants and timepoints. Therefore, two feature tables were used for downstream analysis: non-collapsed and collapsed. The collapsed feature table contains one reference sample per infant and timepoint, obtained by averaging all ASV abundances from that infant at that timepoint. This version is used for correlating behavioural outcome measures, ensuring that over-represented infants do not skew the results. The non-collapsed feature table is used for all other analyses to retain all data and avoid unnecessary information loss (see Figure \ref{fig:workflow_diversity_sig}, \ref{fig:workflow_diversity_outcome}).

\subsection*{Diversity}
Diversity metrics were calculated for both the collapsed and non-collapsed feature tables. A k-mer–based approach was used, which allows assessment of genetic similarity without a computationally intensive phylogenetic reconstruction. \\
To determine an appropriate sampling threshold, rarefaction was performed. Shannon entropy plateaued between 5'000 and 10'000 reads, so a sampling depth of 9'000 reads was chosen to capture community diversity while retaining most samples. Bootstrapping was used to reduce stochastic variation of subsampling \cite{raspet_facilitating_2025}. For the k-mer size selection, bootstrapping was performed for k = 12, 14, and 16. Shannon and Pilou metrics were stable across all lengths, so k = 12 was chosen to increase the likelihood of matches across sequences and improve detection of related taxa. \\
The final diversity analysis was performed with a k-mer size of 12, a sampling depth of 9'000 and 100 iterations.
After the discussion in our presentation, ASV-based diversity estimation was also performed to compare temporal changes in alpha diversity with the k-mer based approach. However, the rest of the analysis is based on k-mer derived diversity.

\subsection*{Inter-Infant differences and temporal trajectories}
To assess inter-infant differences and temporal changes, diversity metrics estimated from the non-collapsed feature table were analyzed (see Figure \ref{fig:workflow_diversity_sig}). 
%\textbf{Statistical Testing Alpha diversity}\\

Differences in alpha diversity between timepoints (2, 4, and 6 months) were assessed using the Kruskal–Wallis test. Shannon diversity was used as the metric, as it accounts for both richness and evenness. The analysis was performed for both k-mer and ASV derived diversity.
%\textbf{PCoA and Statistical Testing Beta diversity}\\

The distance matrices Jaccard and Bray-curtis were examined with PCoA \cite{halko_algorithm_2011}. Visualizations in this report are based on data from infants that had samples at all three timepoints (2, 4 and 6 months).  
Statistical testing of beta diversity on braycurtis distances was performed using Adnois, a multivariate analysis of variance with permutations \cite{martinez_arbizu_pedro_pairwise_nodate}. This approach accounts for the longitudinal study design and controls for repeated sampling per infant as well as the non-independence of samples across timepoints.
R was used to perform this analysis, since the Adonis function in the q2-longitudinal plugin is currently not functional and multilevel sample comparison is not implemented yet \cite{lizgehret_depr_nodate, bokulich_enable_nodate}.

\subsection*{Differential Abundance / Temporal Trajectories}
In order to quantify global temporal trends in microbiome composition in the gut microbiome of infants, ANCOM-BC2 is implemented \cite{lin_multigroup_2024}. 
To control for repeated sampling, a random intercept is fitted for each infant. 
As the SILVA databased used for taxonomy classification does not provide reliable species level information, the taxa were collapsed on genus level.
Taxa that are present in fewer than 10 samples were filtered and a prevalence cutoff of 0.05 was provided to ANCOM-BC2.

Inter-infant variability of these trends is assessed with the feature-volatility action in the longitudinal QIIME plugin \cite{bokulich_q2-longitudinal_2018}.
We train a Random Forest Regressor to identify taxa that are important for predicting the time point of gut microbiome sampling. 
Plotting the relative frequency of these taxa for each infant over time gives insight into infant-specific changes as well as global trends.

PICRUSt2 allows us to predict the functional potential of the gut microbiome based on amplicon data \cite{langille_predictive_2013}.
The resulting pathway abundance are analyzed with ANCOM-BC2 analogous to the method described before.
In order to identify overarching themes of biological activity at the different timepoints, we perform Gene Set Enrichment Analysis as implemented in GSEApy \cite{subramanian_gene_2005,mootha_pgc-1alpha-responsive_2003,fang_gseapy_2023}.
Pathway sets used for this analysis are based on the secondary pathway level of the MetaCyc database \cite{caspi_metacyc_2014}. 
The enrichment rank of each pathway is determined by calculating $-\operatorname{sgn}(\text{LFC}) \log_{10}(q)$ and sorting the set in descending order.
This sorted set is then used for preranked GSEA.

\subsection*{Gut Microbiome Diversity and Behavioural Outcome}
Pearson correlation of the behavioural measures "Behavioural Development", "Sleep Quality", "Sleep Rhythmicity", and "Attenuated Caring Style" with each other and the age of the infants is calculated.
To quantify the contribution of k-mer-based Shannon entropy to the outcome measures, a Mixed Linear Model is fit using QIIME longitudinal \cite{bokulich_q2-longitudinal_2018}.
By controlling for infant age and repeated sampling (infant id), we isolate the effect of the gut microbiome diversity.

\subsection*{Predicting Behaviour with Microbiome Composition}
In order to investigate the influence of microbiome composition on behavioural measures, multiple machine learning models were trained. 
"Behavioural Development" was predicted using center-log transformed abundance data, as recommended for compositional count data \cite{quinn_field_2019,gloor_microbiome_2017}. 
Nested cross-validation with hyperparamter tuning was performed for a Lasso and Gradient Boosted Regressor \cite{caspi_metacyc_2014}. 
To assess generalizing capabilities of the model to unseen timepoints, the cross-validation splits were assigned non-overlapping groups based on infant id and timepoint.
"Sleep Quality" was predicted with the same approach. 
Additionally, pathway abundance as predicted by PICRUSt2 was used as model features.
Performance was assessed by inspecting per-fold performance metrics for train and test set of the outer folds, as well as by calculating pooled performance metrics over all predictions.
%TODO maybe explain here why full feature table?

\begin{figure*}[p]
\centering
\captionsetup[subfigure]{justification=centering}
\begin{subfigure}{\linewidth}
  \centering
  \includegraphics[width=1\linewidth,trim={2cm 8cm 2cm 9.5cm}, clip]{misc/workflow_preprocessing.pdf}
  \caption{Workflow - Preprocessing}
  \label{fig:workflow_preprocessing}
\end{subfigure}

\begin{subfigure}{\linewidth}
  \centering
  \includegraphics[width=1\textwidth,trim={2cm 4.5cm 2cm 6.5cm}, clip]{misc/workflow_diversity_sig.pdf}
  \caption{Workflow - Inter-infant differences and Temporal Trajectories}
  \label{fig:workflow_diversity_sig}
\end{subfigure}

\begin{subfigure}{\linewidth}
  \centering
  \includegraphics[width=1\textwidth,trim={2cm 6.8cm 2cm 8.2cm}, clip]{misc/workflow_outcome_meas.pdf}
  \caption{Workflow - Changes and prediction of behavioural outcome measures}
  \label{fig:workflow_diversity_outcome}
\end{subfigure}
\caption{}
\label{}
\end{figure*}


% \begin{figure*}[ht]
%   \centering
%   \includegraphics[width=1\textwidth,trim={2cm 2.5cm 2cm 2.5cm}, clip]{misc/qc_bases.jpg}
%   \caption{ score per base position}
%   \label{fig:_control}
% \end{figure*}

\section*{Results}
\subsection*{Inter-Infant differences and temporal trajectories}
% \textbf{Statistical Testing Alpha diversity}\\
For the k-mer–based analysis, no significant differences in Shannon diversity were observed between 2, 4, and 6 months, indicating that alpha diversity remains stable across infants during the first six months of life. Interestingly, when looking at the alpha diversity calculated from ASV-based core metrics, there is a significant difference between 2 and 4 months ($q=0.089$) and between 2 and 6 months ($q=0.089$).

% \textbf{PcoA Bray-Curtis}\\
PCoA based on Jaccard distances did not show any visible clustering, whereas PCoA using Bray-Curtis distances did. This indicates that the differences between the samples are driven by changes in relative abundance rather than the presence or absence of specific taxa.\\
PCoA on Bray–Curtis distances revealed distinct clusters by infant, with multiple subclusters forming within individual infants (see Figure X). When samples were labeled by timepoint, no clear clusters or visual patterns were observed (see Figure X). However, considering both infant and timepoint together showed that the subclusters that formed within individual infants corresponded to different timepoints (see Figure X).\\
This first visual inspection indicated that the microbiome of infants differs strongly between each other. Within individual infants, the microbiome appears to change over time. However, when excluding the infant id information, no clear trends are observable across timepoints.

% \textbf{Statistical Testing Beta diversity}\\
Assessment of inter-infant differences using Adonis revealed significant differences between infants ($p<0.001$). Interestingly, significant differences were also observed between timepoints. The pairwise comparisons showed significant differences between 2 and 6 months ($p < 0.001$) and 4 and 6 months ($p<0.001$), while no significant difference was found between 2 and 4 months ($p = 0.295$). This suggests that the microbiome of 6 month old infants is very distinct from that of younger infants.\\
Including both variable into the model revealed that the infant id ($R^2 = 0.64$) explains substantially more of the variance in the distance matrix than timepoint ($R^2 = 0.04$).  
This also quantitatively confirms the observations from the PCoA analysis. Adonis showed that the effect of the infant is so strong that it can mask the effect of timepoint, explaining why no clear clustering by timepoint was observed when infant information was excluded.
% TODO PCAO plots into Subfigures 
\begin{figure}[ht]
  \centering
  \includegraphics[width=0.9\linewidth, clip]{misc/infant_cluserting.pdf}
  \caption{PcoA Bray-Curtis Infant}
  \label{fig:pcoa_infant}
\end{figure}

\begin{figure}[h]
  \centering
  \includegraphics[width=0.9\linewidth, clip]{misc/timepoint_cluserting.pdf}
  \caption{PcoA Bray-Curtis Timepoint}
  \label{fig:pcoa_timepoint}
\end{figure}

\begin{figure}[h]
  \centering
  \includegraphics[width=0.9\linewidth, clip]{misc/timepoint_infant_cluserting.pdf}
  \caption{PcoA Bray-Curtis Infant and Timepoint}
  \label{fig:pcoa_infant_timepoint}
\end{figure}

\subsection*{Differential Abundance / Temporal Trajectories}

*Figure relative abundance* shows clear changes in microbiome composition over time. Enterobacteria and Bifidobacteria decrease slightly in abundance, whereas Veillonella increase in abundance and dominate the gut microbiome of older infants.
Multiple members of the Bacillota phylum are significantly enriched (q < 0.05) at 6 months compared to two months, such as Phascolarctobacterium, Ruminococcus and Enterococcus. 
Other members, such as Clostridum and Staphylococcus, are depleted significantly. 
The gram-negative genus Klebsiella shows significantly lower abundance  at 6 months.

The taxa volatility plot for Veillonella (see Figure) show a general trend of increased abundance from 4 to 6 months. 
Nevertheless, not all infants follow this trend, as multiple do not show increased abundance over time. 
The inverse is true for the Enterobacteriaceae family, as there is a general decrease in abundance but two infants that show contradictory behaviour.

Differential abundance analysis of predicted pathway abundance in the gut microbiome at two and 6 months reveals multiple trends (Figure). 
Acetly-CoA fermentation (PWY-5676) is enriched 2-fold at 6 compared to 2 months. 
Fucose and N-acetylneuraminate degradation is significantly depleted at 6 months. 
Multiple pathways related to gram-negative bacteria (LPS synthesis, O-antigen biosynthesis and enterobacterial common antigen synthesis) are less abundant at 6 months.

GSEA reveals no significantly enriched or depleted pathway sets between 2 and 6 months. 

\subsection*{Gut Microbiome Diversity and Behavioural Outcom}

Figure heatmap shows the correlation and corresponding significance levels of the outcome measures. 
Sleep quality and sleep rhtyhmicity correlate positively with age. 
Sleep rhythmicity also correlates positively with an attenuated caring style of parents.
The behavioural development score correlates negatively with age. 

K-mer based Shannon entropy does not have a significant effect on any outcome measure tested. 
The slope of the relationship between microbiome diversity and behavioural outcome is near zero for three out of four measures (see Figure correlation) with non-significant p-values. 
Only sleep rhythmicity has a moderately positive association with Shannon entropy. 
The same analysis performed with conventional Shannon entropy yields nearly identical results. 

\subsection*{Predicting Behaviour with Microbiome Composition}
% TODO CHAGNE to POOLED metrics

Naive prediction of behavioural development with microbiome composition without taking into account  dependence of samples yields high predictive performance (see Figure). 
The Lasso regression achieves an RMSE of 16.4937 ± 4.5039 and $R^2$ of 0.4632 ± 0.5222 on the test splits.
When taking into account dependence and enforcing generalization to unseen infant - timepoint combinations, performance drops drastically (see Figure).
Test $RMSE$ and $R^2$ increase or drop to 19.9908 ± 13.4954 and -1.0823 ± 1.5417
respectively. 
The gradient boosted regressor performs worse ($RMSE = 27.7251 \pm 14.8527$, $R^2 = -2.1999 \pm 2.9022$) and is plagued by overfitting as indicated by overly optimistic training statistics ($RMSE = 0.0000 \pm 0.0000$, $R^2 = 1.0000 \pm 0.0000$).

\subsection*{Discussion}


\twocolumn[
  \begin{@twocolumnfalse}

%BIBLIOGRAPHY---
\pagebreak
\printbibliography

\end{@twocolumnfalse}
]

\end{document}