\documentclass[a4paper, twocolumn, 10pt]{article}

\usepackage{anysize} % Package to change margin size
\marginsize{2cm}{2cm}{1cm}{2cm}
\usepackage{fancyhdr} % Package to make headers
\renewcommand{\headrulewidth}{0pt}
\usepackage[dvipsnames]{xcolor} % Colors for the references links
\usepackage{hyperref} % Package to link references
\hypersetup{
    colorlinks=true,
    linkcolor=black,
    citecolor=CadetBlue,
    filecolor=CadetBlue,      
    urlcolor=CadetBlue,
}

% \usepackage{float}
\usepackage{graphicx, subcaption, stfloats}
\usepackage{booktabs, multirow}
\usepackage{amsmath}

\usepackage{balance}

\usepackage[style=numeric, sorting=none, url=false, isbn=false, backend=biber]{biblatex}
\addbibresource{./microbEvolve2.bib}

\DeclareBibliographyDriver{misc}{%
  \printnames{author}
  \printfield{title}
  \printfield{url}%
  \printfield{urldate}
  \finentry
}

\renewenvironment{abstract} % Sets abstract
 {\par\noindent\textbf{\abstractname}\ \ignorespaces \\}
 {\par\noindent\medskip}

\begin{document}
% Makes header
\pagestyle{fancy}
% \thispagestyle{empty}
\fancyhead[R]{\includegraphics[height=1cm]{misc/eth_logo_kurz_pos.png}}
\fancyhead[L]{}
% Makes footnotes with an asterisk
\renewcommand*{\thefootnote}{\fnsymbol{footnote}}
    
\twocolumn[
  \begin{@twocolumnfalse}

% TITLE ----
\begin{center}
\Large{\textbf{MicrobEvolve2 -- Final Report}} \\
\vspace{0.4cm}
\normalsize
\begin{tabular}{rl}
  Yara Roth & \texttt{rothya@ethz.ch} \\
  Flurin Schindele & \texttt{fschindele@ethz.ch} \\
  Emelie Guggisberg & \texttt{eguggisberg@ethz.ch}
\end{tabular} \\
\vspace{0.1cm}
% \textit{ETH Zurich}
%\small{Other Location}
\medskip
\normalsize
\end{center}
{\color{gray}\hrule}
\vspace{0.4cm}

% ABSTRACT ----- (if needed)
\begin{abstract}
The gastrointestinal microbiome undergoes rapid development during early life and is influenced by multiple factors such as delivery mode and feeding. Increasing evidence indicates that the early microbiome composition influences not only metabolic, immune, and endocrine functions but also neurocognitive outcomes.

The goal of this project is to analyze the development of microbial diversity and composition in infants during their first 6 months of life and associate changes to behavioural outcome measures using 16S rRNA gene sequencing data.

K-mer-derived diversity estimation shows no significant differences in Shannon diversity between 2, 4, and 6 months. In contrast, ASV-based diversity reveals a significant between Shannon entropy and age ($p=0.022$). Beta diversity based on Bray-Curtis differs significantly between infants and timepoints (both $q < 0.001$). Differential abundance analysis indicates diet-associated differences in predicted pathway abundances. Lastly, no significant correlations are observed between k-mer based Shannon diversity and behavioural outcome measures, and behavioural measures can not be accurately predicted from microbial profiles.
This project confirms that microbial diversity and composition change during the first months of life and highlights the challenge of correlating complex behavioural outcomes with microbiome changes.\\

{\color{gray}\hrule}
\medskip
\end{abstract}
\end{@twocolumnfalse}
]

%\tableofcontents

% TEXT ----

% HINTS ----
% Please use the figure* (or table*) environment instead of figure to include
% page-wide graphics. Figure will use column-wide graphics.

\section*{Introduction}
During the first years of life, the gut microbiome undergoes rapid and extensive development before stabilizing into an adult-like state \cite{milani_first_2017,fahur_bottino_early_2025,yassour_natural_2016}. This rapid development is influenced by multiple factors, including the maternal microbiota, mode of delivery, feeding, and antibiotic exposure \cite{zimmermann_factors_2018,fahur_bottino_early_2025,bokulich_antibiotics_2016}.
The gut microbial communities play a key role in immune, metabolic, and endocrine pathways and therefore directly influence host development \cite{robertson_human_2019,bokulich_antibiotics_2016}. This also explains why altered colonization events in early life are associated with the development of inflammatory disorders, metabolic diseases, and neurocognitive outcomes \cite{fahur_bottino_early_2025}.
Growing evidence suggests a bidirectional relationship between sleep and gut microbiome composition, with many sleep disorders linked to alterations in the microbiome \cite{sen_microbiota_2021}.
Identifying key microbial differences linked to neurocognitive functions, such as sleep or rhythmicity, could uncover potential therapeutic targets, for example, through targeted nutritional interventions to improve sleep.

The goal of this project is to analyze the development of microbial diversity in infants during their first 6 months of life. We also aim to assess compositional changes and explore their functional implications. In a second part, we investigate how microbiome changes relate to behavioural outcome measures and attempt to predict these outcomes from microbial profiles.

We work with data from an observational longitudinal cohort study that followed healthy infants during their first 6 months of life \cite{kerff_gut_2025}. Stool samples were collected at 2, 4, and 6 months, along with behavioural measures. The V4 region of the 16S rRNA gene was sequenced using Illumina NextSeq2000.

\begin{figure*}[p]
\centering
\captionsetup[subfigure]{justification=centering}
\begin{subfigure}{\linewidth}
  \centering
  \includegraphics[width=1\linewidth,trim={2cm 8cm 2cm 9.5cm}, clip]{misc/workflow_preprocessing.pdf}
  \caption{Preprocessing}
  \label{fig:workflow_preprocessing}
\end{subfigure}

\begin{subfigure}{\linewidth}
  \centering
  \includegraphics[width=1\textwidth,trim={2cm 5cm 2cm 5cm}, clip]{misc/workflow_diversity_sig.pdf}
  \caption{Inter-Infant, Inter-Time differences and Temporal Trajectories}
  \label{fig:workflow_diversity_sig}
\end{subfigure}

\begin{subfigure}{\linewidth}
  \centering
  \includegraphics[width=1\textwidth,trim={2cm 7.3cm 2cm 6.8cm}, clip]{misc/workflow_outcome_meas.pdf}
  \caption{Changes and Prediction of Behavioural Outcome Measures}
  \label{fig:workflow_diversity_outcome}
\end{subfigure}
\caption{Workflow Analysis}
\label{}
\end{figure*}

\section*{Methods}
\subsection*{Preprocessing}
Preprocessing of the 16S rRNA amplicon sequencing data includes data importing, quality control, trimming and denoising, taxonomic classification, and feature table preparation (see Figure \ref{fig:workflow_preprocessing}).

First, the demultiplexed sequences, which were provided as QIIME 2 artifacts, are imported along with the corresponding metadata. Quality control is performed next. Forward reads display a median quality score of 34 for all base positions. The reverse reads also display a median of 34, which drops to 20 for the last base positions. The variability is higher for the reverse reads and increases substantially towards the end of the reads. Both forward and reverse sequences are 301 bps long.

The V4 region of the 16S rRNA gene is shorter than 301 bps, which indicates the presence of primers or read-through in the sequences \cite{ezbiocloud_16s_nodate}. To verify the presence of primers, we perform an initial trimming attempt with Cutadapt using V4-specific forward and reverse primer sequences \cite{martin_cutadapt_2011,earthmicrobiome_16s_nodate}. This results in zero sequences being trimmed, meaning that the primers were already removed by the sequencing facility. Another possible explanation for the read length is read-through, which is confirmed by running Cutadapt with the reverse complements of the primers. Approximately 4.5 million sequences are successfully truncated, resulting in reads of $\approx$250 bases, likely representing the true amplicon length. Truncation fails for many reversed reads, likely due to the low base quality at the ends, preventing Cutadapt from recognizing the reverse complement of the forward primer. Therefore, truncation is performed directly during denoising with DADA2 \cite{callahan_dada2_2016}. Forward and reverse reads are truncated to 220 bp and 200 bp, which is sufficient to remove the read-through while maintaining an overlap of $\approx$130 base pairs. This results in good denoising performance, with 90\% of reads passing filtering and nearly all reads successfully being merged.

Taxanomic classification is performed using a weighted classifier optimized for stool samples \cite{bokulich_optimizing_2018}. The classifier targets the 16S rRNA V4 region (515F/806R) and is based on the SILVA 138.2 database (99\% NR) \cite{quast_silva_2013}.

The last step of preprocessing involves preparation of the feature table and metadata. The metadata consists of two files: one with per sample information and one with per-age information, including behavioural outcome measures at each timepoint. Both files are merged into a complete metadata file to simplify further analysis.
The number of samples collected per infant at a given timepoint varies depending on stool frequency, resulting in uneven sampling across infants and timepoints. Therefore, two feature tables are used for downstream analysis: non-collapsed and collapsed. The collapsed feature table contains one reference sample per infant and timepoint, obtained by averaging all ASV abundances from an infant at a timepoint. This version is used for associating behavioural outcome measures to microbiome data, ensuring that overrepresented infants do not skew the results. The non-collapsed feature table is used for all other analyses to avoid unnecessary information loss (see Figure \ref{fig:workflow_diversity_sig}, \ref{fig:workflow_diversity_outcome}).

\subsection*{Diversity}
Diversity metrics are calculated for both the collapsed and non-collapsed feature tables. A k-mer-based approach is used, which allows assessment of genetic similarity without a computationally intensive phylogenetic reconstruction. To determine an appropriate sampling threshold, rarefaction is performed. Shannon entropy plateaus between 5'000 and 10'000 reads, so a sampling depth of 9'000 reads is chosen to capture community diversity while retaining most samples. Bootstrapping was used to reduce stochastic variation of subsampling \cite{raspet_facilitating_2025}. For the k-mer size selection, bootstrapping is performed for k = 12, 14, and 16. Shannon and Pilou metrics are stable across all lengths, so k = 12 is selected to increase the sensitivity to detect closely related taxa.
The final diversity analysis is performed with a k-mer size of 12, a sampling depth of 9'000, and 100 bootstrapping iterations.

After the discussion in our presentation, ASV-based diversity estimation is also performed to compare temporal changes in alpha diversity with the k-mer-based approach. However, the rest of the analysis is based on k-mer-derived diversity.

\subsection*{Inter-Infant and Inter-Timepoint Differences}
To assess inter-infant and inter-timepoint differences, diversity metrics estimated from the non-collapsed feature table are analyzed (see Figure \ref{fig:workflow_diversity_sig}). 

Differences in alpha diversity between timepoints (2, 4, and 6 months) are assessed using the Kruskal-Wallis test. 
Correlation of alpha diversity and age is assessed by Spearman correlation.
Shannon diversity is used as the metric, as it accounts for both richness and evenness.

The distance matrices Jaccard and Bray-Curtis are examined with PCoA \cite{halko_algorithm_2011}. Visualizations in this report are based on data from infants that had samples at all three timepoints (2, 4, and 6 months).  

Statistical testing of beta diversity using Bray-Curtis distances is performed with Adonis, a permutation-based multivariate analysis of variance. \cite{martinez_arbizu_pedro_pairwise_nodate}.
This approach accounts for the dependence of samples across timepoints due to the longitudinal study design and controls for repeated sampling per infant.
R was used to perform this analysis since the Adonis function in the q2-longitudinal plugin is currently not functional and multilevel sample comparison is not implemented yet \cite{lizgehret_depr_nodate, bokulich_enable_nodate}.

\subsection*{Temporal Trajectories in the Infant Gut Microbiome}
To quantify global temporal trends in microbiome composition and explore their functional implications, differential abundance and functional annotation are performed on the non-collapsed feature table (see Figure \ref{fig:workflow_diversity_sig}).

Differential abundance is performed using ANCOM-BC2. \cite{lin_multigroup_2024}. 
To control for repeated sampling, a random intercept is fitted for each infant. 
As the SILVA database used for taxonomy classification does not provide reliable species level information, the taxa are collapsed on genus level.
Taxa that are present in fewer than 10 samples were filtered and a prevalence cutoff of 0.05 was provided to ANCOM-BC2.

Inter-infant variability of these trends is assessed with the feature-volatility action in the longitudinal QIIME plugin \cite{bokulich_q2-longitudinal_2018}.
We train a Random Forest Regressor to identify taxa that are important for predicting the time point of gut microbiome sampling. 
Plotting the relative frequency of these taxa for each infant over time provides insight into infant-specific changes as well as global trends.

PICRUSt2 allows us to predict the functional potential of the gut microbiome based on amplicon data \cite{langille_predictive_2013}.
The resulting pathway abundance is analyzed with ANCOM-BC2, analogous to the method described before.
To identify overarching themes of biological activity at the different timepoints, we perform gene set enrichment analysis as implemented in GSEApy \cite{subramanian_gene_2005,mootha_pgc-1alpha-responsive_2003,fang_gseapy_2023}.
Pathway sets used for this analysis are based on the secondary pathway level of the MetaCyc database \cite{caspi_metacyc_2014}. 
The enrichment rank of each pathway is determined by calculating $-\operatorname{sgn}(\text{LFC}) \log_{10}(q)$ and sorting the set in descending order.
This sorted set is then used for preranked GSEA.

\subsection*{Gut Microbiome Diversity and Behavioural Outcome}

The dependence of each behavioural measure on other factors is assessed by looking into correlations. 
Pearsons correlations are calculated between the behavioural outcome measures (``Behavioural Development'', ``Sleep Quality'', ``Sleep Rhythmicity'', and ``Attenuated Caring Style'') and with the age of the infants.
To quantify the contribution of k-mer-based Shannon entropy to the outcome measures, a Mixed Linear Model is fit using QIIME longitudinal \cite{bokulich_q2-longitudinal_2018}.
By controlling for infant age and repeated sampling (infant id), we isolate the effect of the gut microbiome diversity.

\subsection*{Predicting Behaviour with Microbiome Composition}
To investigate the influence of microbiome composition on behavioural measures, multiple machine learning models are trained. 
``Behavioural Development'' is predicted using center-log transformed abundance data, as recommended for compositional count data \cite{quinn_field_2019,gloor_microbiome_2017}. 
Nested cross-validation with hyperparamter tuning is performed for a Lasso and Gradient Boosted Regressor \cite{caspi_metacyc_2014}. 
To assess generalizing capabilities of the model to unseen timepoints, the cross-validation splits were assigned non-overlapping groups based on infant id and timepoint.
``Sleep Quality'' is predicted with the same approach. 
Additionally, pathway abundance, as predicted by PICRUSt2, is used as model features.
Performance was assessed by inspecting per-fold performance metrics for train and test set of the outer folds, as well as by calculating pooled performance metrics over all predictions.
%TODO maybe explain here why full feature table?

\begin{figure}[ptb]
\centering
\captionsetup[subfigure]{justification=centering}
\begin{subfigure}{0.95\linewidth}
  \centering
  \includegraphics[width=0.9\linewidth, clip]{misc/infant_cluserting.pdf}
  \caption{Infant Clustering}
  \label{fig:pcoa_infant}
\end{subfigure}

\begin{subfigure}{0.95\linewidth}
  \centering
  \includegraphics[width=0.9\linewidth, clip]{misc/timepoint_cluserting.pdf}
  \caption{Timepoint CLustering}
  \label{fig:pcoa_timepoint}
\end{subfigure}

\begin{subfigure}{0.95\linewidth}
  \centering
  \includegraphics[width=0.9\linewidth, clip]{misc/timepoint_infant_cluserting.pdf}
  \caption{Infant and Timepoint CLustering}
  \label{fig:pcoa_infant_timepoint}
\end{subfigure}

    \caption{PCoA based on Bray-Curtis}
    \label{fig:}
\end{figure}


\section*{Results}

\subsection*{Inter-Infant and Inter-Timepoint Differences}
For the k-mer-based analysis, no significant differences in Shannon diversity are observed between 2, 4, and 6 months, and Spearman correlation with age is not significant.
This indicates that alpha diversity remains stable across infants during the first 6 months of life. Interestingly, when looking at the alpha diversity calculated from ASV-based core metrics, there is a significant correlation between Shannon entropy and age ($p = 0.022$).

PCoA based on Jaccard distances do not show any visible clustering, whereas PCoA using Bray-Curtis distances does. This indicates that the differences between the samples are driven by changes in relative abundance rather than the presence or absence of specific taxa.

PCoA on Bray-Curtis distances reveals distinct clusters by infant, with multiple subclusters forming within individual infants (see Figure \ref{fig:pcoa_infant}). When samples are labeled by timepoint, no clear clusters or visual patterns can be observed (see Figure \ref{fig:pcoa_timepoint}). However, considering both infant and timepoint together shows that the subclusters that form within individual infants correspond to different timepoints (see Figure \ref{fig:pcoa_infant_timepoint}).
This first visual inspection indicats that the microbiome of infants differs strongly between each other. Within individual infants, the microbiome appears to change over time. However, when excluding the infant id information, no clear trends are observable across timepoints.

Assessment of inter-infant differences in beta diversity using Adonis reveals significant differences between infants and timepoints (both $p<0.001$). The pairwise comparisons show significant differences between 2 and 6 months ($p < 0.001$) and between 4 and 6 months ($p<0.001$), while no significant difference is found between 2 and 4 months ($p = 0.295$). This suggests that the microbiome of 6-month-old infants is very distinct from that of younger infants.
Including both variables in the model reveals that the infant id ($R^2 = 0.64$) explains substantially more of the variance in the distance matrix than the timepoint ($R^2 = 0.04$).  
This also quantitatively confirms the observations from the PCoA analysis. Adonis shows that the effect of the infant is so strong that it can mask the effect of timepoint, explaining why no clear clustering by timepoint is observed when infant information is excluded.

\subsection*{Temporal Trajectories in the Infant Gut Microbiome}

Figure \ref{fig:relabundancefamilies} shows clear changes in microbiome composition over time. Enterobacteria and Bifidobacteria decrease slightly in abundance, whereas Veillonella increase in abundance and dominate the gut microbiome of older infants.
Multiple members of the Bacillota phylum are significantly enriched  ($q < 0.05$) as indicated by ANCOM-BC2 at 6 months compared to 2 months, such as Phascolarctobacterium, Ruminococcus and Enterococcus. 
Other members, such as Clostridum and Staphylococcus, are depleted significantly.
The gram-negative genus Klebsiella shows significantly lower abundance at 6 months.
Surprisingly, Veillonella do not show a significant difference in abundance.

The taxa volatility plot for Veillonella show a general trend of increased abundance from 4 to 6 months. 
Nevertheless, not all infants follow this trend, as multiple do not show increased abundance over time. 
The inverse is true for the Enterobacteriaceae family, as there is a general decrease in abundance but two infants that show contradictory behaviour.
This again highlights the high individualitiy of the infant gut microbiome.

Differential abundance analysis of predicted pathway abundance in the gut microbiome at 2 and 6 months reveals multiple trends (see Figure \ref{fig:pathwaydiffabundance}).
These results need to be interpreted cautiously, as predicting functional potential with PICRUSt2 introduces additional noise and uncertainty which might not reflect the biological reality.
Acetly-CoA fermentation (PWY-5676) is enriched 2-fold at 6 compared to 2 months. 
This pathway is associated with short chain fatty acid digestion that might stem from fiber-containing foods.
Fucose and N-acetylneuraminate degradation, which is linked to a breast-milk focused diet, is significantly depleted at 6 months. 
Multiple pathways related to gram-negative bacteria (LPS synthesis, O-antigen biosynthesis and enterobacterial common antigen synthesis) are less abundant at 6 months.
GSEA reveals no significantly enriched or depleted pathway sets between 2 and 6 months (see Figure \ref{fig:gseapathway}).
This can be attributed to the high-level pathway grouping, as only level two pathway sets from MetaCyc are available to us.
Subtle changes in biological niches can be overpowered due to the number of pathways in a set.

\begin{figure*}[tp]
  \centering
  \includegraphics[width=0.5\linewidth]{../../figures/family_relative_abundance_over_time.pdf}
  \caption{Relative abundance of families ($>0.01$ abundance) over time}
  \label{fig:relabundancefamilies}
\end{figure*}

\begin{figure*}[tbp]
  \centering
  \includegraphics[width=0.8\linewidth]{../../figures/differential_pathway_abundance_6months_vs_2months.pdf}
  \caption{Differential abundance of selected pathways comparing 2 to 6 months}
  \label{fig:pathwaydiffabundance}
\end{figure*}

\begin{figure*}[tbp]
  \centering
  \includegraphics[width=0.8\linewidth]{../../figures/gsea_metacyc_pathways_second_level.pdf}
  \caption{Enrichment of secondary level MetaCyc pathways sets}
  \label{fig:gseapathway}
\end{figure*}

\subsection*{Gut Microbiome Diversity and Behavioural Outcome}

Figure \ref{fig:corrheatmap} shows the correlation and corresponding significance levels of the outcome measures. 
Sleep quality and sleep rhythmicity correlate positively with age. 
This is to be expected, as most parents can tell a story about newborns sleeping irregularly.
Sleep rhythmicity also correlates positively with an attenuated caring style of parents.
The behavioural development score correlates negatively with age. 

K-mer based Shannon entropy does not have a significant effect on any outcome measure tested. 
Figure \ref{fig:regshannon} does not show the results of the MLE model but rather a simple regression fitted on the data and only acts as a visual aid for the following results.
The slope of the relationship between microbiome diversity and behavioural outcome is near zero for sleep quality and caring style with non-significant p-values ($p = 0.633,\; 0.904$ respectively).
The effect of Shannon entropy on behavioural development is slightly stronger ($\beta = 18.524$) but not significant ($p = 0.133$).
Sleep rhythmicity has a moderately positive association with Shannon entropy ($\beta = 0.113$, $p = 0.052$). 
The same analysis performed with ASV-derived Shannon entropy yields nearly identical results.

\begin{figure}[tb]
  \centering
  \includegraphics[width=\linewidth]{../../figures/outcome_measures_correlation_matrix.pdf}
  \caption{Heatmap showing Pearson correlation}
  \label{fig:corrheatmap}
\end{figure}

\begin{figure*}[pb]
  \centering
  \includegraphics[width=0.7\linewidth]{../../figures/shannon_entropy_vs_outcome_measures.pdf}
  \caption{Regression plot for Shannon entropy and behavioural measures.}
  \label{fig:regshannon}
\end{figure*}

\subsection*{Predicting Behaviour with Microbiome Composition}

Naive prediction of behavioural development with microbiome composition without considering dependence of samples yields good predictive performance (see Figure \ref{fig:pred_behaviour_naive_lasso}, pooled $R^2 = 0.69$, $RMSE = 17.01$). 
The Lasso regression achieves a $RMSE$ of $16.49 \pm 4.50$ and $R^2$ of $0.46 \pm 0.52$ on the test splits.
When considering dependence and enforcing generalization to unseen infant - timepoint combinations, performance drops drastically (see Figure \ref{fig:pred_behaviour_generalize_lasso}).
Pooled $R^2$ and $RMSE$ drops and increases to 0.05 and 29.49, respectively.
The gradient boosted regressor performs worse (pooled $R^2 = -0.20$, $RMSE = 33.19$) and is plagued by overfitting as indicated by overly optimistic training statistics (per-fold $RMSE = 0.00 \pm 0.00$, $R^2 = 1.00 \pm 0.00$). 
This indicates strong overfitting of the model on the training data which does not allow for generalization to the test data.
The higher internal complexity of this model compared to a Lasso Regression has the advantage of potentially picking up abstract information in the data but increases the risk of overfitting, which has happened here.

Prediction of sleep qualitity follows similar results.
Generalizing to unseen timepoints of infants leads to very limited predictive performance (see Figure \ref{fig:pred_sleep_microbiome_lasso}).
Using predicted pathway abundance as the model features decreases performance slightly compared to the raw microbiome composition data.
The model trained on pathway abundance achieves a pooled $R^2 = 0.08$ and $RMSE = 0.13$ (see Figure \ref{fig:pred_sleep_pathway_lasso}).
We hypothesized that providing pathway abundance as model features would make biologically relevant information more easily accessible to the model.
Pathway abundance might relate more directly to physiological outcomes such as sleep quality and behavioural development.
Due to the relatively small training data set size, making relevant information explicitly available is crucial. 
In contrast, large models trained on more data might be able to infer abstract patters in the data.
Poorer performance can also be explained in part by the introduction of additional noise and uncertainty through the functional prediction with PICRUSt2.
The gradient boosted regressor performs worse than Lasso regression ($R^2=-0.30$, $RMSE=0.15$).
Again, overfitting is the main cause of this poor performance.

\begin{figure*}[tbp]
\centering
\captionsetup[subfigure]{justification=centering}
\begin{subfigure}{0.475\textwidth}
    \centering
    \includegraphics[width=\linewidth, height=6cm, keepaspectratio=true]{../../figures/pred_behaviour_naive_lasso.pdf} 
    \caption{}
    \label{fig:pred_behaviour_naive_lasso}
\end{subfigure}
\begin{subfigure}{0.475\textwidth}
    \centering
    \includegraphics[width=\linewidth, height=6cm, keepaspectratio=true]{../../figures/pred_behaviour_generalize_lasso.pdf}
    \caption{}
    \label{fig:pred_behaviour_generalize_lasso}
\end{subfigure}

\begin{subfigure}{0.475\textwidth}
    \centering
    \includegraphics[width=\linewidth, height=6cm, keepaspectratio=true]{../../figures/pred_sleep_microbiome_lasso.pdf} 
    \caption{}
    \label{fig:pred_sleep_microbiome_lasso}
\end{subfigure}
\begin{subfigure}{0.475\textwidth}
    \centering
    \includegraphics[width=\linewidth, height=6cm, keepaspectratio=true]{../../figures/pred_sleep_pathway_lasso.pdf}
    \caption{}
    \label{fig:pred_sleep_pathway_lasso}
\end{subfigure}

    \caption{Predicting Behavioural Development and Sleep Quality with Microbiome Data}
    \label{fig:ml}
\end{figure*}

\subsection*{Discussion}

The results of this project align well with the existing literature
We were able to confirm an increase in alpha diversity of the infant gut microbiome over time when looking at ASV-based Shannon entropy.
The discrepancy with k-mer-based diversity could be linked to the genetic relationship between taxa.
Even though richness and evenness of taxa increases, they might be closely related and therefore share many k-mers, resulting in non-significant differences in the k-mer-based approach.

Beta-diversity analysis has revealed that the gut microbiome seems to have a distinct state at 6 months.
Dirichlet multinomial mixtures modeling of infant gut microbiome samples has also shown 6 months as a breaking point for clusters, underlining our results \cite{stewart_temporal_2018}. This might be linked to changes in infant diet, as weaning and introduction of more complex foods conventionally starts at this time.
Changes in diet go hand in hand with colonization of the gut by taxa that find new metabolic niches. 
We were able to observe this by looking into predicted pathway abundance. 
From 2 to 6 months, pathways associated with human milk oligosaccharide (HMO) breakdown, such as fucose and acetlyneuraminate degradation, decreased in abundance, which is most likely associated with weaning \cite{yu_comprehensive_2025,duman_human_2024}.
The increase in Acetly-CoA fermentation can be linked to the introduction of solid foods that contain undigestable fibers.

\balance

Differential abundance analysis of microbiome composition did not reproduce hallmark results that have been reported in the literature.
For example, we did not observe a globally significant decrease in Bifidobacteria or Enterobacteria \cite{koenig_succession_2011}.
This contrasts the findings regarding HMO degradation, as Bifidobacteria are one of the main drivers of this phenomenon.

A major factor influencing all analysis performed in this project is the individuality of the infant gut microbiome.
Dimensionality reduction and Adonis have revealed clear clustering of samples from the same infant.
Controlling for dependence of samples has therefore been a crucial pillar of our work.
This was further complicated by gaps in infant sampling, resulting in only five infants having been sampled at all three timepoints.
Missing entire timepoints from multiple infants makes it increasingly difficult to identify global trends over time.
Despite the difficulties concerning patient sampling, stricter sampling and a larger cohort would enhance future work.

Additionally, the issue with limited sample size when investigating highly complex and multifactorial neurocognitive outcomes was illustrated by poor predictive performance of machine learning models.
Even though the gut microbiome has previously been linked to such outcomes and biologically reasonable assumptions exist, the relationship is with high likelihood complex and messy \cite{sen_microbiota_2021}.
More sophisticated optimization and potentially hypothesis-driven feature enrichment in combination with large sample sizes might enable many more discoveries.

In conclusion, this work confirmed that microbial diversity and composition change during the first months of life. Associations between behavioural outcome measures and microbiome changes were challenging to detect, underscoring the difficulty of linking such multifactorial measures to the microbiome.

\subsection*{Code and Data Availability}

All code for the analysis mentioned in this report and additional work can be found on \href{https://github.com/Emelie23/microbEvolve2}{GitHub}.

\clearpage
% \twocolumn[
%   \begin{@twocolumnfalse}

%BIBLIOGRAPHY---

\printbibliography

% \end{@twocolumnfalse}
% ]

\end{document}